Documentation/Patterns.md

```diff
@@ -0,0 +1,77 @@
+# SerenityOS patterns
+
+----
+## Introduction
+
+Overtime numerous reoccurring patterns have emerged from or where adopted by
+the serenity code base. This document aims to track and describe them so they
+can be propagated further and keep the code base consistent. 
+
+----
+## Intrusive Lists
```

Comment:
Might be worth saying briefly what an intrusive list is and why it's useful. It's pretty easy to look up the phrase online but I hadn't heard of it before joining Serenity.

Reply:
Sure, I added a little chunk, let me know what you think.

Reply:
:+1:

Though I think it should be "the elements membership" -> "the element's membership".

Reply:
done